packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -48,6 +48,29 @@ class SlashCommandRoutingType(HandlerRoutingType):
     underscores."""
 
 
+class MarkdownHelpFormatter(argparse.ArgumentDefaultsHelpFormatter):
+    def _get_help_string(self, action):
+        # do not show "(default: False)" for flags as this is assumed
+        if action.const is True:
+            return action.help
+        return super()._get_help_string(action)
+
+    def _format_action_invocation(self, action):
+        if not action.option_strings:
+            return super._format_action_invocation(action)
+        else:
+            option_strings = [f"`{string}`" for string in action.option_strings]
+            if action.nargs == 0:
+                return ", ".join(option_strings)
+            else:
+                default = self._get_default_metavar_for_optional(action)
+                args_string = self._format_args(action, default)
+                return ", ".join(option_strings) + " " + args_string
+
+    def _format_action(self, action):
+        return "- " + super()._format_action(action)
+
```

Comment:
I have some optional suggestions for improving the help string rendered to the user, especially for `/learn -h`:

1. Can you add a delimiter (e.g. `:`) between each argument and its help text? 
2. Can you ensure each argument's help string is always capitalized in Title-case?
3. For each argument, can the argument syntax be surrounded in a single inline Markdown block?


This doesn't need to be addressed here, since this PR already makes a significant improvement to the user experience as-is. However, if it is easy to do, then it is worth implementing here.

Reply:
When calling `/learn -h` for instance, instead of printing:

```md
`-h`, `--help` show this help message
...
`-r`, `--remote` REMOTE learn a remote document; currently only arxiv is supported (default: None)
```

I would like to see:

```md
`-h/--help`: Show this help message
...
`-r/--remote REMOTE`: Learn a remote document; currently only arxiv is supported (default: None)
```

Reply:
Screenshot for reference:

<img width="600" alt="Screenshot 2024-07-08 at 10 33 22 AM" src="https://github.com/jupyterlab/jupyter-ai/assets/44106031/79e91724-d7b9-4762-b4a2-8f6b428402f6">

Reply:
> 1. Can you add a delimiter (e.g. :) between each argument and its help text?

Makes sense! The `/help` uses a dash (`—`), should we use it for consistency over a colon (`:`)? Then the colon is shorter which is a big advantage here.

https://github.com/jupyterlab/jupyter-ai/blob/4664a0baa2993efb2b0265d41f376dd5c902c6c8/packages/jupyter-ai/jupyter_ai/chat_handlers/help.py#L30

> 2. Can you ensure each argument's help string is always capitalized in Title-case?

Will do! For reference, the current state (prior to this PR) was using lowercase which reflects the default capitalization in `argparse` (for example see [here](https://docs.python.org/3/library/argparse.html#example)) On the other hand, the `/help` uses Title case. I lean only slightly towards lowercase as it seems easier to parse.

> 3. For each argument, can the argument syntax be surrounded in a single inline Markdown block?

Sure thing, thank you for the example!

Reply:
> The /help uses a dash (—), should we use it for consistency over a colon (:)?

I prefer a colon since the emdash looks extremely similar to a hyphen, which shows up in every slash command option. So using a colon as a delimiter here would be easier to read.